### PR TITLE
add eleven strategy for lp

### PR DESCRIPTION
**Pull request by midnight-commit**

### CODE REVIEW

File: contracts/strategies/ElevenStrategyForLPV1.sol [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        _stakeDepositTokens(amount);
+        emit Deposit(account, amount);
+    }
+
+    function withdraw(uint amount) external override {
+        require(amount > 0, "ElevenStrategyV1::withdraw");
+        uint elevenShares = _convertSharesToElevenShares(amount);
+        stakingContract.withdraw(PID, elevenShares);
+        vaultContract.withdrawAll();
+        uint depositTokenAmount = depositToken.balanceOf(address(this));
```
Comment (ASlowCheetah) [inline contracts/strategies/ElevenStrategyForLPV1.sol:101]: is it possible to have dust of depositToken here?

File: contracts/strategies/ElevenStrategyForLPV1.sol [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        }
+        require(depositToken.transferFrom(msg.sender, address(this), amount), "ElevenStrategyV1::transfer failed");
+        _mint(account, getSharesForDepositTokens(amount));
+        _stakeDepositTokens(amount);
+        emit Deposit(account, amount);
+    }
+
+    function withdraw(uint amount) external override {
+        require(amount > 0, "ElevenStrategyV1::withdraw");
+        uint elevenShares = _convertSharesToElevenShares(amount);
```
Comment (ASlowCheetah) [inline contracts/strategies/ElevenStrategyForLPV1.sol:98]: does this consider the withdrawal fee?

File: contracts/strategies/ElevenStrategyForLPV1.sol [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        stakingContract.withdraw(PID, elevenShares);
+        vaultContract.withdrawAll();
```
Comment (ASlowCheetah) [inline contracts/strategies/ElevenStrategyForLPV1.sol:100]: sorry to ask about how this works, I'm not familiar with Eleven, will the withdrawAll only withdraw the depositTokens you have available from the previous call? if so, is there any chance deposit tokens are left as dust in the contract calls?

**Review by ASlowCheetah — COMMENTED** [review]
Great work!

File: contracts/strategies/ElevenStrategyForLPV1.sol (hunk 3 above)
Comment (snow-farmer) [reply contracts/strategies/ElevenStrategyForLPV1.sol:119]: `withdrawAll` basically checks balance and calls `withdraw(amount)`. We could do these steps ourselves and remove the L120 balanceOf call, which saves gas and could be more secure

File: contracts/strategies/ElevenStrategyForLPV1.sol [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+            pendingReward,
+            address(rewardToken), address(WAVAX),
+            swapPairWAVAXELE
+        );
+    }
+    
+    function _stakeDepositTokens(uint amount) private {
+        require(amount > 0, "ElevenStrategyV1::_stakeDepositTokens");
+        vaultContract.deposit(amount);
+        uint elevenShares = vaultContract.balanceOf(address(this));
```
Comment (snow-farmer) [inline contracts/strategies/ElevenStrategyForLPV1.sol:186]: *

File: contracts/strategies/ElevenStrategyForLPV1.sol (hunk 2 above)
Comment (snow-farmer) [reply contracts/strategies/ElevenStrategyForLPV1.sol:117]: No - withdraw fee is taken from vaultContract withdraw (see L119)

**Pull request merged**